diff --git a/node_modules/react-native-screens/ios/RNSScreen.h b/node_modules/react-native-screens/ios/RNSScreen.h
index 9b83b3a..3275a5b 100644
--- a/node_modules/react-native-screens/ios/RNSScreen.h
+++ b/node_modules/react-native-screens/ios/RNSScreen.h
@@ -139,11 +139,15 @@
 - (void)rns_updateProps:(NSArray<NSString *> *)props;
 
 #ifdef RCT_NEW_ARCH_ENABLED
-- (RCTSurfaceTouchHandler *)rnscreens_findTouchHandlerInAncestorChain;
+#if RCT_NEW_ARCH_ENABLED
+- (RCTSurfaceTouchHandler *)rnscreens_findTouchHandlerInAncestorChain;
 #else
-
+- (RCTTouchHandler *)rnscreens_findTouchHandlerInAncestorChain;
 #endif
 
 @end
 
 @interface RNSScreenManager : RCTViewManager <RCTInvalidating>
diff --git a/node_modules/react-native-screens/ios/RNSScreen.mm b/node_modules/react-native-screens/ios/RNSScreen.mm
index 451c23f..f7d6a7d 100644
--- a/node_modules/react-native-screens/ios/RNSScreen.mm
+++ b/node_modules/react-native-screens/ios/RNSScreen.mm
@@ -627,11 +627,15 @@
 }
 
 #ifdef RCT_NEW_ARCH_ENABLED
-- (RCTSurfaceTouchHandler *)rnscreens_findTouchHandlerInAncestorChain
+#if RCT_NEW_ARCH_ENABLED
+- (RCTSurfaceTouchHandler *)rnscreens_findTouchHandlerInAncestorChain
+#else
+- (RCTTouchHandler *)rnscreens_findTouchHandlerInAncestorChain
 #endif
 {
-  return _touchHandler;
+#if RCT_NEW_ARCH_ENABLED
+  return (RCTSurfaceTouchHandler *)_touchHandler;
+#else
+  return (RCTTouchHandler *)_touchHandler;
+#endif
 }
 
 - (void)didUpdateReactSubviews
 {
